pkg/server/apiextensions.go

```diff
@@ -382,6 +382,31 @@ func shallowCopyCRD(in *apiextensionsv1.CustomResourceDefinition) *apiextensions
 	return &out
 }
 
+// decorateCRDWithBinding copy and mutate crd by
+// 1. adding identity annotation
+// 2. terminating status when apibinding is deleting
+func decorateCRDWithBinding(in *apiextensionsv1.CustomResourceDefinition, identity string, deleteTime *metav1.Time) *apiextensionsv1.CustomResourceDefinition {
+	out := shallowCopyCRD(in)
+
+	out.Annotations[apisv1alpha1.AnnotationAPIIdentityKey] = identity
+
+	if deleteTime.IsZero() {
+		return out
+	}
+
+	out.Status.Conditions = make([]apiextensionsv1.CustomResourceDefinitionCondition, len(in.Status.Conditions), len(in.Status.Conditions)+1)
```

Comment:
these conditions here are not visible to anybody, are they? They are only faked while feeding the apiextensions-apiserver CR registry.

Reply:
nvmd. The Terminating condition is used by apiextensions. Odd. Maybe add a comment to clarify that. Using a condition always feel odd to me for any kind of critical logic.